src/__tests__/dialog-test.tsx

```diff
@@ -158,9 +158,9 @@ test('Closes a dialog on click outside', async () => {
     expect(await screen.findByRole('dialog')).toBeInTheDocument();
     expect(await screen.findByRole('button', {name: 'Nope!'})).toBeInTheDocument();
 
-    await userEvent.click(screen.getByTestId('dialog-overlay'));
+    userEvent.click(screen.getByTestId('dialog-overlay'));
```

Comment:
why this? afaik, `userEvent.click()` returns a promise: https://testing-library.com/docs/user-event/convenience#click

Reply:
If I wait for the click action to finish, the element is removed too early and the following `waitForElementToBeRemoved` fails because the element must exist to wait for removal.